GiniVision/Classes/Core/Camera.swift

```diff
@@ -177,7 +177,9 @@ extension Camera: AVCaptureMetadataOutputObjectsDelegate {
         if let metadataObj = metadataObjects.first as? AVMetadataMachineReadableCodeObject,
             metadataObj.type == AVMetadataObjectTypeQRCode {
             let qrDocument = GiniQRCodeDocument(scannedString: metadataObj.stringValue)
-            didDetectQR?(qrDocument) 
+            DispatchQueue.main.async { [weak self] in
```

Comment:
Is `captureOutput` now called on a secondary thread?

Reply:
And it was called before, but since I changed the logic behind `didDetectQR`, I prefer to execute it in the main thread here and not inside of the closure itself.